include/class.ticket.php

```diff
@@ -2645,14 +2645,17 @@ function update($vars, &$errors) {
             $errors['err'] = __('Missing or invalid data - check the errors and try again');
 
         if ($vars['duedate']) {
+            //Set Due Date/Time and Current Time to DBTimezone
+            $tempduedate = new DateTime($cfg->getDbTimezone().$vars['duedate'].' '.$vars['time']);
+            $currentdate = new DateTime('@'.time());
+            $currentdate->setTimeZone(new DateTimeZone($cfg->getDbTimezone()));
```

Comment:
I like the approach. It seems like this should go with the other time mangling functions in the Misc class